src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2086,22 +2169,89 @@ ErrorIfSubqueryJoin(Query *queryTree)
 
 
 /*
- * MultiSubqueryPushdownTable creates a MultiTable from the given subquery range
- * table entry and returns it. Note that this sets subquery field of MultiTable
- * to subquery of the given range table entry.
+ * PopulateSubqueryTargetEntryAndColumnNameList creates a target entry for each
+ * unique column and populate column names list.
+ */
+static void
+PopulateSubqueryTargetEntryAndColumnNameList(List *uniqueColumnList,
+											 List **subqueryTargetEntryList,
+											 List **columnNamesList)
+{
+	AttrNumber resNo = 1;
+
+	ListCell *columnCell = NULL;
+	foreach(columnCell, uniqueColumnList)
+	{
+		Var *column = (Var *) lfirst(columnCell);
+		TargetEntry *newTargetEntry = makeNode(TargetEntry);
+		StringInfo columnNameString = makeStringInfo();
+
+		newTargetEntry->expr = (Expr *) copyObject(column);
+		appendStringInfo(columnNameString, WORKER_COLUMN_FORMAT, resNo);
+		newTargetEntry->resname = columnNameString->data;
+		newTargetEntry->resjunk = false;
+		newTargetEntry->resno = resNo;
+
+		*subqueryTargetEntryList = lappend(*subqueryTargetEntryList, newTargetEntry);
+		*columnNamesList = lappend(*columnNamesList, makeString(columnNameString->data));
+		resNo++;
+	}
+}
+
+
+/*
+ * UpdateVarMappingsForExtendedOpNode updates varno/varattno fields of columns
+ * in columnList to point to corresponding location in uniqueColumnList. The
+ * subquery target entry list is also created in similar manner to make sure
+ * varattno field from column list matches target entry lists resno field.
+ */
+static void
+UpdateVarMappingsForExtendedOpNode(List *columnList, List *uniqueColumnList)
```

Comment:
I'm noting what we've talked in person. It probably would make easier to read the code once we pass `subqueryTargetEntryList` instead of `uniqueColumnList`. I think this is my final comment.